linux/software/libcamera/csi-2-usage.md

```diff
@@ -0,0 +1,166 @@
+
+Information on this bridge chip can be found on the [Toshiba Website](https://toshiba.semicon-storage.com/ap-en/semiconductor/product/interface-bridge-ics-for-mobile-peripheral-devices/hdmir-interface-bridge-ics/detail.TC358743XBG.html)
+
+The TC358743 interfaces HDMI in to CSI-2 and I2S outputs. It is supported by the [TC358743 kernel module](https://github.com/raspberrypi/linux/blob/rpi-5.4.y/drivers/media/i2c/tc358743.c).
+
+The chip supports incoming HDMI signals as either RGB888, YUV444, or YUV422, at up to 1080p60. It can forward RGB888, or convert it to YUV444 or YUV422, and convert either way between YUV444 and YUV422. Only RGB888 and YUV422 support has been tested. When using 2 CSI-2 lanes, the maximum rates that can be supported are 1080p30 as RGB888, or 1080p50 as YUV422. When using 4 lanes on a Compute Module, 1080p60 can be received in either format.
+
+HDMI negotiates the resolution by a receiving device advertising an [EDID](https://en.wikipedia.org/wiki/Extended_Display_Identification_Data) of all the modes that it can support. The kernel driver has no knowledge of the resolutions, frame rates, or formats that you wish to receive, therefore it is up to the user to provide a suitable file.
+This is done via the VIDIOC_S_EDID ioctl, or more easily using `v4l2-ctl --fix-edid-checksums --set-edid=file=filename.txt` (adding the --fix-edid-checksums option means that you don't have to get the checksum values correct in the source file). Generating the required EDID file (a textual hexdump of a binary EDID file) is not too onerous, and there are tools available to generate them, but it is beyond the scope of this page.
+
+As described above, use the `DV_TIMINGS` ioctls to configure the driver to match the incoming video. The easiest approach for this is to use the command `v4l2-ctl --set-dv-bt-timings query`. The driver does support generating the SOURCE_CHANGED events should you wish to write an application to handle a changing source. Changing the output pixel format is achieved by setting it via VIDIOC_S_FMT, however only the pixel format field will be updated as the resolution is configured by the dv timings.
+
+There are a couple of commercially available boards that connect this chip to the Raspberry Pi. The Auvidea B101 and B102 are the most widely obtainable, but other equivalent boards are available.
+
+This driver is loaded using the `config.txt` dtoverlay `tc358743`.
+
+The chip also supports capturing stereo HDMI audio via I2S. The Auvidea boards break the relevant signals out onto a header, which can be connected to the Pi's 40 pin header. The required wiring is:
+
+| Signal   | B101 header | Pi 40 pin header |
+|----------|:-----------:|:----------------:|
+| LRCK/WFS |     7       |      19          |
+| BCK/SCK  |     6       |      18          |
+| DATA/SD  |     5       |      20          |
+| GND      |     8       |      39          |
```

Comment:
Oops, my bad here. Those are BCM GPIO numbers, not 40pin header pin numbers. Another column needed
```
| Signal   | B101 header | Pi 40 pin header | BCM GPIO  |
|----------|:-----------:|:----------------:|:---------:|
| LRCK/WFS |     7       |       35         |    19     |
| BCK/SCK  |     6       |       12         |    18     |
| DATA/SD  |     5       |       38         |    20     |
| GND      |     8       |       39         |    N/A    |